src/libexpr/eval.cc

```diff
@@ -258,14 +259,23 @@ PosIdx Value::determinePos(const PosIdx pos) const
 
 bool Value::isTrivial() const
 {
-    return
-        internalType != tApp
-        && internalType != tPrimOpApp
-        && (internalType != tThunk
-            || (dynamic_cast<ExprAttrs *>(thunk.expr)
-                && ((ExprAttrs *) thunk.expr)->dynamicAttrs.empty())
-            || dynamic_cast<ExprLambda *>(thunk.expr)
-            || dynamic_cast<ExprList *>(thunk.expr));
+    if (internalType == tApp || internalType == tPrimOp) {
+        return true;
```

Comment:
This clause isn't equivalent to the first two conjuncts in the original expression. Isn't it supposed to be?

Reply:
Whoops, thanks for catching that. Fixed up into aacc0af.